### PR TITLE
fix(ecmascript): Various function names and lengths were wrong, typeof operator didn't handle unresolvable references

### PR DESCRIPTION
**Pull request by aapoalas**
Just copy-paste leftovers, with the exception of `%ThrowTypeError%`.

Included is also a fix to `typeof X` not correctly returning `undefined` if `X` cannot be resolved.

### CODE REVIEW

File: nova_vm/src/ecmascript/builtins/fundamental_objects/function_objects/function_prototype.rs [hunk 1]
```diff
@@ -283,6 +288,52 @@ impl ThrowTypeError {
...
+        },
+    };
+    let (keys, values) = agent
+        .heap
+        .elements
+        .create_object_entries(&[length_entry, name_entry]);
+
+    agent.heap.create(ObjectHeapData {
+        // The value of the [[Extensible]] internal slot of this function is false.
+        extensible: true,
```
Comment (andreubotella) [inline nova_vm/src/ecmascript/builtins/fundamental_objects/function_objects/function_prototype.rs:334]:
```suggestion
        extensible: false,
```

File: nova_vm/src/ecmascript/builtins/numbers_and_dates/date_objects/date_constructor.rs [hunk 2]
```diff
@@ -31,7 +31,7 @@ pub struct DateConstructor;
 impl Builtin for DateConstructor {
     const BEHAVIOUR: Behaviour = Behaviour::Constructor(Self::behaviour);
     const LENGTH: u8 = 1;
```
Comment (andreubotella) [inline nova_vm/src/ecmascript/builtins/numbers_and_dates/date_objects/date_constructor.rs:33]:
```suggestion
    const LENGTH: u8 = 7;
```

**Review by andreubotella — DISMISSED** [review]

File: nova_vm/src/ecmascript/builtins/fundamental_objects/function_objects/function_prototype.rs (hunk 1 above)
Comment (aapoalas) [reply nova_vm/src/ecmascript/builtins/fundamental_objects/function_objects/function_prototype.rs:334]: Oooops :D

File: nova_vm/src/ecmascript/builtins/numbers_and_dates/date_objects/date_constructor.rs (hunk 2 above)
Comment (aapoalas) [reply nova_vm/src/ecmascript/builtins/numbers_and_dates/date_objects/date_constructor.rs]: Huh, that's seemingly missing a test.

**Comment by aapoalas** [comment]
Expectations changes are because I built the file "from the ground up". Lots of crashes have turned to failures.

**Review by andreubotella — APPROVED** [review]

**Pull request merged**